### ISSUE TITLE
Allow TypeScript enums to be linked with AutoConfig

### ISSUE THREAD

**Issue by Nayni** [prisma/nexus#118]
I'm currently struggling with using and `enum` defined in TypeScript which is used as members for an `enumType` in nexus.

The problem is as follows:

I have an existing TypeScript enum: 
```typescript
export enum Category {
    CATEGORY_A = "CATEGORY_A ",
    CATEGORY_B = "CATEGORY_B ",
    CATEGORY_C = "CATEGORY_C ",
    CATEGORY_D = "CATEGORY_D "
}
```

The values of this enum will also be exposed in the schema and thus I created the following `enumType`:
```typescript
export const CategoryEnum = enumType({
    name: "Category",
    members: [
        Category .CATEGORY_A ,
        Category .CATEGORY_B ,
        Category .CATEGORY_C ,
        Category .CATEGORY_D 
    ]
});
```

This works fine, and nexus will understand that my enums are strings and will generate the schema correctly. It will also generate the type definitions for this and they will be:
```typescript
export interface NexusGenEnums {
  Category: "CATEGORY_A " | "CATEGORY_B " | "CATEGORY_C " | "CATEGORY_D "
}
```

And this is where the problem comes into play. Because now when I setup a resolver the incoming type of for example an input will be mapped to the nexus generated type which is a string union type. However in our implementation we always refer to our own TypeScript enum and the generated nexus type doesn't fit into this type because strings do not magically typecheck into an enum (the other way around works fine but not when the input is a string).

I also tried to use the `typegenAutoConfig` and point nexus to our existing enum but it doesn't seem to pick up the alias and keeps using the string union. (even if I change the Regex to include looking for enum)

Is this an expected behavior and if so, would it maybe be a useful feature to also allow TypeScript enums to be configurable as candidates for type alias?

PS: Love the library btw, been working on migrating our codebase and it's been super enjoyable so far.

**Comment by Nayni** [comment]
Did some fiddling with my own fork and came up with the following to allow enums to have backing types (only tested in isolation so far):
https://github.com/prisma/nexus/compare/develop...Nayni:allow-enum-backing-types

Is this something that could be considered @tgriesser ? I'm not sure why enums were excluded from having backing types in the first place?

**Comment by tgriesser** [comment]
Hmm, yeah I don't know why it wasn't considered - probably just an oversight. Feel free to open a PR and I'll review it a little closer. I think we might want this to exclude `const enum`s if possiblee?

**Comment by Nayni** [comment]
Solved by #142 
Thanks for help with the PR!
